src/oemof/solph/models.py

```diff
@@ -362,3 +362,306 @@ def _add_parent_block_variables(self):
+        blocks.MultiObjectiveFlow
+    ]
+
+    def __init__(self, energysystem, **kwargs):
+        super().__init__(energysystem, **kwargs)
+
+    def _add_parent_block_sets(self):
+        """"""
+        # set with all nodes
+        self.NODES = po.Set(initialize=[n for n in self.es.nodes])
+
+        # pyomo set for timesteps of optimization problem
+        self.TIMESTEPS = po.Set(
+            initialize=range(len(self.es.timeindex)), ordered=True
+        )
+
+        # previous timesteps
+        previous_timesteps = [x - 1 for x in self.TIMESTEPS]
+        previous_timesteps[0] = self.TIMESTEPS.last()
+
+        self.previous_timesteps = dict(zip(self.TIMESTEPS, previous_timesteps))
+
+        # pyomo set for all flows in the energy system graph
+        self.FLOWS = po.Set(
+            initialize=self.flows.keys(), ordered=True, dimen=2
+        )
+
+        self.BIDIRECTIONAL_FLOWS = po.Set(
+            initialize=[
+                k
+                for (k, v) in self.flows.items()
+                if hasattr(v, "bidirectional")
+            ],
+            ordered=True,
+            dimen=2,
+            within=self.FLOWS,
+        )
+
+        self.UNIDIRECTIONAL_FLOWS = po.Set(
+            initialize=[
+                k
+                for (k, v) in self.flows.items()
+                if not hasattr(v, "bidirectional")
+            ],
+            ordered=True,
+            dimen=2,
+            within=self.FLOWS,
+        )
+
+    def _add_parent_block_variables(self):
+        """"""
+
+        self.flow = po.Var(self.FLOWS, self.TIMESTEPS, within=po.Reals)
+
+        for (o, i) in self.FLOWS:
+            if self.flows[o, i].nominal_value is not None:
+                if self.flows[o, i].fix[self.TIMESTEPS[1]] is not None:
+                    for t in self.TIMESTEPS:
+                        self.flow[o, i, t].value = (
+                            self.flows[o, i].fix[t]
+                            * self.flows[o, i].nominal_value
+                        )
+                        self.flow[o, i, t].fix()
+                else:
+                    for t in self.TIMESTEPS:
+                        self.flow[o, i, t].setub(
+                            self.flows[o, i].max[t]
+                            * self.flows[o, i].nominal_value
+                        )
+
+                    if not self.flows[o, i].nonconvex:
+                        for t in self.TIMESTEPS:
+                            self.flow[o, i, t].setlb(
+                                self.flows[o, i].min[t]
+                                * self.flows[o, i].nominal_value
+                            )
+                    elif (o, i) in self.UNIDIRECTIONAL_FLOWS:
+                        for t in self.TIMESTEPS:
+                            self.flow[o, i, t].setlb(0)
+            else:
+                if (o, i) in self.UNIDIRECTIONAL_FLOWS:
+                    for t in self.TIMESTEPS:
+                        self.flow[o, i, t].setlb(0)
```

Comment:
Did you actually change something compared to the BaseModel class? If not, we would not need this code, because the methods are provided by the BaseClass, aren't they? (long time doing sub-classing coding in Python)

Reply:
It is an addition compared to the BaseModel class, yes, but not compared to the Model class. So I think the MultiObjectiveModel class should inherit from the Model class instead of the BaseModel class to reduce code duplication.

Reply:
Hi, quite a nice feature and a bit related to the pending multiperiod feature #750.
It seems, the only difference here is amending the CONSTRAINT_GROUPS list by block.MultiObjectiveFlow. So letting the class inherit from the Model class seems like a good idea and would save you redefining the methods already given there.

Reply:
The dependency is now fixed, together with a few linting errors. Codacy throws an error because of an `import` statement which it should ignore. I don't really know what to do there, suggestions are highly appreciated. 
Other than that, all pipelines are passed.

Altering the `AUTHORS.rst` and `CHANGELOG.rst` created a merge conflict, so I reverted that commit. I will try to alter those files one by one, to see whether only one of them creates a merge conflict and then go on from there.